util/logtools/encounter_tools.ts

```diff
@@ -286,3 +286,191 @@ export class EncounterCollector extends EncounterFinder {
     this.lastSeal = undefined;
   }
 }
+
+class TLUtilFunctions {
+  timeFromDate(date: Date | undefined): string {
+    if (date)
+      return date.toISOString().slice(11, 19);
+    return 'Unknown_Time';
+  }
```

Comment:
Here and for most of the other string-slice-and-dice functions for filenames, I figured that since we have type information available on our `EncounterInfo` types, we might as well handle potentially undefined properties within the utility functions themselves rather than attempting to massage the data before passing it to the functions. That is, I feel like it makes more sense to do this:

```typescript
const obj: SomeEncounterInfo = { SomeData };
const date = TLUtilFuncs.dayFromDate(obj.startTime); // We know that startTime is either Date or undefined already, so just use it
```

As opposed to 
```typescript
const obj: SomeEncounterInfo = { SomeData };
let date; // We have no idea whether startTime is defined, so we have to check for it explicitly, oh no.
if (obj.startTime)
    date = TLUtilFuncs.dayFromDate(obj.startTime);
else
    date = 'Unknown_Date';
```